File: android/app/src/main/java/com/izzatismail/midtrim/domain/usecase/CalculateMergedDurationUseCase.kt
package com.izzatismail.midtrim.domain.usecase

class CalculateMergedDurationUseCase {
    fun execute(trimDuration: Double, videoCount: Int): Double {
        if (trimDuration < 0 || videoCount < 0) return 0.0
        return trimDuration * videoCount
    }
}

File: ios/MidTrim/MidTrim/Domain/UseCases/CalculateMergedDurationUseCase.swift
import Foundation

struct CalculateMergedDurationUseCase {
    func execute(trimDuration: Double, videoCount: Int) -> Double {
        guard trimDuration >= 0, videoCount >= 0 else {
            return 0
        }
        return trimDuration * Double(videoCount)
    }
}
